src/test/java/edu/ucsb/cs156/example/controllers/IceCreamShopControllerTests.java

```diff
@@ -0,0 +1,305 @@
+        @Test
+        public void admin_can_delete_a_iceCreamShop() throws Exception {
+                // arrange
+
+                IceCreamShop baskinrobbins = IceCreamShop.builder()
+                                .name("BaskinRobbins")
+                                .address("StateStSantaBarbaraCA")
+                                .description("Colorfulicecreamparlorchainknownforitsmanyflavorsplussorbetyogurt")
+                                .build();
+
+                when(iceCreamShopRepository.findById(eq(15L))).thenReturn(Optional.of(baskinrobbins));
+
+                // act
+                MvcResult response = mockMvc.perform(
+                                delete("/api/icecreamshop?id=15")
+                                                .with(csrf()))
+                                .andExpect(status().isOk()).andReturn();
+
+                // assert
+                verify(iceCreamShopRepository, times(1)).findById(15L);
+                verify(iceCreamShopRepository, times(1)).delete(any());
+
+                Map<String, Object> json = responseToJson(response);
+                assertEquals("IceCreamShop with id 15 deleted", json.get("message"));
+        }
+
+        @WithMockUser(roles = { "ADMIN", "USER" })
+        @Test
+        public void admin_tries_to_delete_non_existant_iceCreamShop_and_gets_right_error_message()
+                        throws Exception {
+                // arrange
+
+                when(iceCreamShopRepository.findById(eq(15L))).thenReturn(Optional.empty());
+
+                // act
+                MvcResult response = mockMvc.perform(
+                                delete("/api/icecreamshop?id=15")
+                                                .with(csrf()))
+                                .andExpect(status().isNotFound()).andReturn();
+
+                // assert
+                verify(iceCreamShopRepository, times(1)).findById(15L);
+                Map<String, Object> json = responseToJson(response);
+                assertEquals("IceCreamShop with id 15 not found", json.get("message"));
+        }
+
+        @WithMockUser(roles = { "ADMIN", "USER" })
+        @Test
+        public void admin_can_edit_an_existing_iceCreamShop() throws Exception {
+                // arrange
+
+                IceCreamShop ivdripOrig = IceCreamShop.builder()
+                                .name("IVDrip")
+                                .address("EmbarcaderodelNorteIslaVistaCA")
+                                .description("Quaintcompactcafeservinglocallyroastedcoffeealongsidehousemadebakedtreatsicecream")
+                                .build();
+
+                IceCreamShop ivdripEdited = IceCreamShop.builder()
+                                .name("ivdrip")
+                                .address("EmbarcaderodelNorteGoletaCA")
+                                .description("Quaintcompactcafeservinglocallyroastedcoffeealongsidehousemadebakedtreatsicecream")
```

Comment:
The description set in this line should probably be different from the one in `ivdripOrig`, so that it doesn't leave a mutation alive from pitest.

Reply:
Updated!